pkg/discovery/worker/container_spec_discovery_worker.go

```diff
@@ -0,0 +1,87 @@
+	}
+	return containerSpecEntityDTOs, nil
+}
+
+// getContainerDataAggregator returns utilization and usage data aggregator based on the given utilization and usage data
+// aggregation strategies. If given data aggregation strategies are not supported, use default strategies.
+func (worker *k8sContainerSpecDiscoveryWorker) getContainerDataAggregator(utilizationDataAggStrategy,
+	usageDataAggStrategy string) (agg.ContainerUtilizationDataAggregator, agg.ContainerUsageDataAggregator) {
+	utilizationDataAggregator, exists := agg.ContainerUtilizationDataAggregators[utilizationDataAggStrategy]
+	if !exists {
+		glog.Errorf("Container utilization data aggregation strategy %s is not supported. Use default % strategy",
+			utilizationDataAggStrategy, agg.DefaultContainerUtilizationDataAggStrategy)
+		utilizationDataAggregator = agg.ContainerUtilizationDataAggregators[agg.DefaultContainerUtilizationDataAggStrategy]
+	}
+	glog.Infof("ContainerSpec will aggregate Containers utilization data by %s", utilizationDataAggregator.AggregationStrategy())
+
+	usageDataAggregator, exists := agg.ContainerUsageDataAggregators[usageDataAggStrategy]
+	if !exists {
+		glog.Errorf("Container usage data aggregation strategy %s is not supported. Use default % strategy",
+			utilizationDataAggStrategy, agg.DefaultContainerUsageDataAggStrategy)
+		usageDataAggregator = agg.ContainerUsageDataAggregators[agg.DefaultContainerUsageDataAggStrategy]
+	}
+	glog.Infof("ContainerSpec will aggregate Containers usage data by %s", usageDataAggregator.AggregationStrategy())
+	return utilizationDataAggregator, usageDataAggregator
+}
+
+// createContainerSpecMap creates map from ContainerSpec ID to ContainerSpec entity with merged VCPU and VMem commodity
+// DTOs of container replicas.
+func (worker *k8sContainerSpecDiscoveryWorker) createContainerSpecMap(containerSpecList []*repository.ContainerSpec) map[string]*repository.ContainerSpec {
+	// Map from ContainerSpec ID to ContainerSpec object
+	containerSpecMap := make(map[string]*repository.ContainerSpec)
+	for _, containerSpec := range containerSpecList {
+		containerSpecId := containerSpec.ContainerSpecId
+		existingContainerSpec, exists := containerSpecMap[containerSpecId]
+		if !exists {
+			containerSpecMap[containerSpecId] = containerSpec
+		} else {
+			// Append commodity DTOs of the same commodity type of container replicas
+			for commodityType, existingCommodities := range existingContainerSpec.ContainerCommodities {
+				commodities, exists := containerSpec.ContainerCommodities[commodityType]
+				if !exists {
+					glog.Errorf("%s commodities do not exist for ContainerSpec %s", commodityType, containerSpec.ContainerSpecId)
+					continue
+				}
+				existingCommodities = append(existingCommodities, commodities...)
+				existingContainerSpec.ContainerCommodities[commodityType] = existingCommodities
+			}
+			// Increment number of container replicas
+			existingContainerSpec.ContainerReplicas++
```

Comment:
Maybe we could log the number of container replicas for a spec. I'm not sure we're using this ContainerReplicas count anywhere.